datafusion/sql/src/parser.rs

```diff
@@ -561,17 +577,13 @@ impl<'a> DFParser<'a> {
                 if token == Token::EOF || token == Token::SemiColon {
                     break;
                 } else {
-                    return Err(ParserError::ParserError(format!(
-                        "Unexpected token {token}"
-                    )));
+                    return parser_err!(format!("Unexpected token {token}"))?;
```

Comment:
The parser_err macro already handles the string interpolation so you can do this insted:

```suggestion
                    return parser_err!("Unexpected token {token}")?;
```

(There are a few other examples of this below)

Reply:
we can probably use `expected` function here. I will try to commit that and other
suggestions tomorrow. (It's quite late here)

Reply:
No rush -- I think we are going to hold of merging any more code changes until we make a 47.0.0 release branch / candidate.